lib/spider/scripts/in_context_learning.py

```diff
@@ -91,19 +119,39 @@ def main(cfg) -> None:
                     data_point["schema"] = schema_file.read()
             else:
                 data_point["schema"] = ""
-            prompt = cfg.prompts.template.format(data_point["schema"], data_point["question"])
+            if cfg.prompts.template.count("{}") > 2:
+                dev_question = data_point["question"]
+                with torch.no_grad():
+                    dev_embedding = encoder_model.encode(dev_question, convert_to_tensor=True)
+
+                cos_similarities = F.cosine_similarity(dev_embedding, train_embeddings, dim=1)
+
+                top_results = torch.topk(cos_similarities, k=top_k)
+
+                similar_questions = [train_questions[index] for index in top_results.indices]
+                format_args = []
+                for element in train_json:
+                    if element["question"] in similar_questions:
+                        format_args.append(element["question"])
+                        format_args.append(element["query"])
+                format_args.append(data_point["schema"])
+                format_args.append(data_point["question"])
+                prompt = cfg.prompts.template.format(*format_args)
+
+            else:
+                prompt = cfg.prompts.template.format(data_point["schema"], data_point["question"])
             messages = [
                 {"role": "user", "content": prompt},
             ]
             encodeds = tokenizer.apply_chat_template(messages, return_tensors="pt")
             model_inputs = encodeds.to("cuda")
-            outputs = model.generate(
-                model_inputs,
-                generation_config=generation_config,
-                pad_token_id=tokenizer.eos_token_id,
-            )
-
-            response = tokenizer.decode(outputs[0], skip_special_tokens=True).split("[/INST]")[1]
+            with torch.no_grad():
+                outputs = model.generate(
+                    model_inputs,
+                    generation_config=generation_config,
+                    pad_token_id=tokenizer.eos_token_id,
+                )
+            response = tokenizer.decode(outputs[0], skip_special_tokens=True)
```

Comment:
Why the removal of `.split()`? (I don't really know why it was introduced either, btw)

Reply:
I think it was introduced to separate the prompt from the actual answer, but I don't really know why we don't need it anymore here